### PR TITLE
ddl :fix cancel add/drop partitioned table ddl job

### PR DESCRIPTION
**Pull request by ciscoxll**
### What problem does this PR solve? 
We can only cancel add partitioned table job when the job state is on JobStateNone , if the job status is JobStateDone or JobStateRunning, the job will not be canceled.

### What is changed and how it works?

Only added tests.
### Check List 

Tests 

 - Unit test

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8938)

### CODE REVIEW

**Comment by ciscoxll** [comment]
@winkyao @zimulala @crazycs520 PTAL.

**Comment by ciscoxll** [comment]
@crazycs520  @winkyao @zimulala PTAL.

File: ddl/rollingback.go [hunk 1]
```diff
@@ -160,6 +160,20 @@ func rollingbackAddindex(w *worker, d *ddlCtx, t *meta.Meta, job *model.Job) (ve
 	return
 }
 
+func rollingbackAddTablePartition(t *meta.Meta, job *model.Job) (ver int64, err error) {
```
Comment (crazycs520) [inline ddl/rollingback.go]: can we only return error? remove ver?

File: ddl/rollingback.go [hunk 2]
```diff
@@ -175,6 +189,20 @@ func rollingbackDropTableOrView(t *meta.Meta, job *model.Job) error {
 	return nil
 }
 
+func rollingbackDropTablePartition(t *meta.Meta, job *model.Job) (ver int64, err error) {
```
Comment (crazycs520) [inline ddl/rollingback.go]: ditto.

File: ddl/rollingback.go [hunk 3]
```diff
@@ -175,6 +189,20 @@ func rollingbackDropTableOrView(t *meta.Meta, job *model.Job) error {
...
+	tblInfo, _, err := checkTablePartitionExist(t, job)
+	if err != nil {
+		return ver, errors.Trace(err)
+	}
+	// Here drop table partition is done in a transaction, if the job is not completed, it can be canceled.
+	if tblInfo.State == model.StatePublic {
```
Comment (crazycs520) [inline ddl/rollingback.go]: Actually, this check will always be true, right?

**Comment by winkyao** [comment]
@ciscoxll Please address comments.

**Comment by ciscoxll** [comment]
@zimulala @crazycs520 PTAL.

**Comment by ciscoxll** [comment]
@winkyao PTAL

File: ddl/rollingback.go [hunk 4]
```diff
@@ -160,6 +160,21 @@ func rollingbackAddindex(w *worker, d *ddlCtx, t *meta.Meta, job *model.Job) (ve
...
 }
 
+func rollingbackAddTablePartition(t *meta.Meta, job *model.Job) error {
+	_, err := getTableInfoAndCancelFaultJob(t, job, job.SchemaID)
+	if err != nil {
+		job.State = model.JobStateCancelled
+		return errors.Trace(err)
+	}
+	// If the job is not started, tidb can cancel this job.
+	if job.SchemaState == model.StateNone {
```
Comment (crazycs520) [inline ddl/rollingback.go]: Use `cancelOnlyNotHandledJob` to remove redundant code?

File: ddl/rollingback.go [hunk 5]
```diff
@@ -175,6 +190,20 @@ func rollingbackDropTableOrView(t *meta.Meta, job *model.Job) error {
 	return nil
 }
 
+func rollingbackDropTablePartition(t *meta.Meta, job *model.Job) error {
+	_, err := checkTableExistAndCancelNonExistJob(t, job, job.SchemaID)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	// If the job is not started, tidb can cancel this job.
+	if job.SchemaState == model.StateNone {
```
Comment (crazycs520) [inline ddl/rollingback.go]: ditto.

**Comment by ciscoxll** [comment]
@crazycs520 @zimulala PTAL.

File: ddl/rollingback.go [hunk 6]
```diff
@@ -230,12 +246,16 @@ func convertJob2RollbackJob(w *worker, d *ddlCtx, t *meta.Meta, job *model.Job)
 		ver, err = rollingbackAddColumn(t, job)
 	case model.ActionAddIndex:
 		ver, err = rollingbackAddindex(w, d, t, job)
+	case model.ActionAddTablePartition:
```
Comment (winkyao) [inline ddl/rollingback.go]: add model.ActionAddTablePartition and model.ActionDropTablePartition type in the admin.isJobRollbackable  function?

Comment (ciscoxll) [reply ddl/rollingback.go]: Done.

**Pull request merged**